client/my-sites/checkout/composite-checkout/lib/paypal-express-processor.ts

```diff
@@ -27,10 +27,7 @@ export default async function payPalProcessor(
 		siteSlug,
 		contactDetails,
 	} = transactionOptions;
-	recordTransactionBeginAnalytics( {
-		reduxDispatch,
-		paymentMethodId: 'paypal',
-	} );
+	reduxDispatch( recordTransactionBeginAnalytics( { paymentMethodId: 'paypal' } ) );
```

Comment:
By the way, this now shows up as a TypeScript error:

```
Argument of type '(dispatch: Dispatch) => void' is not assignable to parameter of type 'Action<any>'.
```

I assume that means that the type for `reduxDispatch` cannot handle thunks?

Reply:
That's true, because the type is specified as `ReturnType< typeof useDispatch >`, which are by default just the vanilla `{ type }` Redux actions, for stores without any middlewares.

Once your store has middlewares that allows dispatching other things, too (like `redux-thunk`), you need to declare your own type.

I'm doing that in #57821 by introducing `CalypsoDispatch` type and using it everywhere where it belongs.

Reply:
Thank you!